Makefile

```diff
@@ -20,23 +18,31 @@ ci: get_tools get_vendor_deps install test_cover test_lint test
 ########################################
 ### Build/Install
 
-check-ledger:
+# Add ledger support - Makefile variable changes are out of a target's scope. There is no point in putting this into a target.
 ifeq ($(LEDGER_ENABLED),true)
-   	ifeq ($(UNAME_S),OpenBSD)
-   		$(info "OpenBSD detected, disabling ledger support (https://github.com/cosmos/cosmos-sdk/issues/1988)")
-TMP_BUILD_TAGS := $(BUILD_TAGS)
-BUILD_TAGS = $(filter-out ledger, $(TMP_BUILD_TAGS))
-   	else
-   	   	ifndef GCC
-   	   	   $(error "gcc not installed for ledger support, please install or set LEDGER_ENABLED to false in the Makefile")
-   	   	endif
-   	endif
-else
-TMP_BUILD_TAGS := $(BUILD_TAGS)
-BUILD_TAGS = $(filter-out ledger, $(TMP_BUILD_TAGS))
+  ifeq ($(OS),Windows_NT)
+    GCCEXE = $(shell where gcc.exe 2> NUL)
+    ifeq ($(GCCEXE),)
+      $(error gcc.exe not installed for ledger support, please install or set LEDGER_ENABLED=false)
+    else
+      BUILD_TAGS += ledger
+    endif
+  else
+    UNAME_S = $(shell uname -s)
+    ifeq ($(UNAME_S),OpenBSD)
+      $(warning OpenBSD detected, disabling ledger support (https://github.com/cosmos/cosmos-sdk/issues/1988))
+    else
+      GCC = $(shell command -v gcc 2> /dev/null)
+      ifeq ($(GCC),)
+        $(error gcc not installed for ledger support, please install or set LEDGER_ENABLED=false)
+      else
+        BUILD_TAGS += ledger
+      endif
+    endif
+  endif
 endif
 
-build: check-ledger update_gaia_lite_docs
```

Comment:
no check anymore?

Reply:
Nope! That's the beauty of it. Check-ledger becomes an intrinsic check at every run (since the change of Makefile variables is out of target scope). The `update_gaia_lite_docs` should not run together with `build` since they have complete different goals, they are part of a different process.

Reply:
Do we want to check it every run though? In many cases the user might not care and doesn't need to see the warning.